node.js

```diff
@@ -97,15 +96,15 @@ Node.prototype.addChild = function (node) {
   return this
 }
 
-Node.prototype.reset = function (prefix) {
+Node.prototype.reset = function (prefix, versions) {
   this.prefix = prefix
   this.children = {}
   this.kind = this.types.STATIC
   this.handlers = new Handlers()
   this.numberOfChildren = 0
   this.regex = null
   this.wildcardChild = null
-  this.versions = SemVerStore()
+  this.versions = versions
```

Comment:
Could we instead have `this.versions = this.versioning.storage()`? This way we keep all of the reset logic in this same function?

Reply:
hi @IpaliboWhyte,

I don't see how this may be possible, since `this.versioning` is part of the `Router` (`index.js`) instance. While the line you commented on is part of the `Node` (`node.js`) instance. 

What we can do, is to pass `versions` store constructor to the `Node` constructor, so we can use it afterwards. In this case `Node` constructor will both accept instantiated version of the `versions` store AND store constructor - so we are increasing the complexity here, a little.

Do you think we should still do that?

Reply:
Hi @PavelPolyakov, You are right! I missed the fact that `versioning` property is of `Router`. So no need to bother with that change for now :)